### PR TITLE
Fix autounexile looping on fail

### PR DESCRIPTION
**Pull request by OTCompa**
Ticket: MOD-14
This PR fixes a bug where autounexile did not remove exile entries upon failure and now additionally properly handles cases where the user left during the timed exile. It also fixes a minor typo in the ban command.

[MWAY-26]: https://naurffxiv.atlassian.net/browse/MWAY-26?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/workers/autounexile.py [hunk 1]
```diff
@@ -16,15 +16,21 @@ async def autounexile_users(self):
...
+                    raise Exception(error_message)
 
-        exiles_database.remove_user_exiles(
-            exile.user_id
-        )  # remove entry from database no matter what
+                error_message = await unexile_user(autounexile_embed, member)
+            if error_message is not None:
+                raise Exception(error_message)
+        except Exception:
+            logger.info(f"Auto Unexile failed, removing exile entry {exile.exile_id}")
```
Comment (KingPieson) [inline src/workers/autounexile.py:35]: Can we also log the user's discordID here? In case moderation wants to manually interact with the user

Comment (brtran4) [reply src/workers/autounexile.py:35]: we should also leave a message for mods to make sure that the roles look correct

Comment (OTCompa) [reply src/workers/autounexile.py]: The embed should already have the user's discord ID but I also added it to the logger just in case!

**Review by KingPieson — APPROVED** [review]

File: src/workers/helper.py [hunk 2]
```diff
@@ -2,17 +2,18 @@
...
 from settings import get_settings
 from util import create_interaction_embed_context
+from typing import Optional
 
 settings = get_settings()
 
 
 def create_autounexile_embed(
-    self, user: discord.User, exile_id: str, end_timestamp: str
+    self, user: Optional[discord.Member], discord_id, exile_id: str, end_timestamp: str
```
Comment (brtran4) [inline src/workers/helper.py:11]: nit: since all the other vars are typed, we might as well type `discord_id`

**Review by brtran4 — CHANGES_REQUESTED** [review]

**Review by brtran4 — APPROVED** [review]
i'm hesitant to press approve because of the new requirement that we don't drop exiles, but i think we should do it in https://naurffxiv.atlassian.net/browse/MOD-40. so mini approve?

**Comment by clairefleuret** [comment]
This should be merged as a temp fix even if we are not going to be dropping exiles in the future.

**Pull request merged**